pkg/util/docker/network.go

```diff
@@ -137,6 +137,60 @@ func resolveDockerNetworks(containerNetworks map[string][]dockerNetwork) {
 
 // DefaultGateway returns the default Docker gateway.
 func DefaultGateway() (net.IP, error) {
+	fields, err := defaultGatewayFields()
+	if err != nil || len(fields) < 3 {
+		return nil, err
+	}
+
+	ipInt, err := strconv.ParseUint(fields[2], 16, 32)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse ip %s from route file: %s", fields[2], err)
+	}
+	ip := make(net.IP, 4)
+	binary.LittleEndian.PutUint32(ip, uint32(ipInt))
+	return ip, nil
+}
+
+// DefaultHostIPs returns the IP addresses bound to the default network interface.
+// The default network interface is the one connected to the network gateway, and it is determined
+// by parsing the routing table file in the proc file system.
+func DefaultHostIPs() ([]string, error) {
+	fields, err := defaultGatewayFields()
+	if err != nil {
+		return nil, err
+	}
+	if len(fields) == 0 {
+		return nil, fmt.Errorf("missing interface information from routing file")
+	}
```

Comment:
IMO `defaultGatewayFields()` should return an error in this case, but the original code contain tests asserting this behavior.